File: 运动物体跟踪CShop/运动物体跟踪CShop/FileOperation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;
using System.Windows.Forms;
using System.Drawing;

namespace 运动物体跟踪CShop
{
    class FileOperation
    {
        static public void writeToFile(string filePath)
        {
            FileStream fs = new FileStream(filePath, FileMode.OpenOrCreate);
            StreamWriter sw = new StreamWriter(fs);
            sw.Write(Global.eventList.Count.ToString() + " " + Global.jiange.ToString() + " " + Global.fps.ToString());
            sw.Write("\r\n");
            for (int i = 0; i < Global.eventList.Count; i++)
            {
                EventNode node = Global.eventList[i];
                int startFrame = node.startFrame;
                int endFrame = node.endFrame;
                sw.Write(startFrame.ToString() + " " + endFrame.ToString() + " " + node.trackList.Count.ToString() + " ");
                for (int j = 0; j < node.trackList.Count; j++)
                {
                    int x = node.trackList[j].X;
                    int y = node.trackList[j].Y;
                    int width = node.trackList[j].Width;
                    int height = node.trackList[j].Height;
                    sw.Write(x.ToString() + " " + y.ToString() + " " + width.ToString() + " " + height.ToString() + " ");
                }
                sw.Write("\r\n");
            }
            sw.Flush();
            sw.Close();
            fs.Close();
        }

        static public void readFromFile(string filePath)
        {
            int N = 3;
            StreamReader fileReader = new StreamReader(filePath);
            string head = fileReader.ReadLine();
            string[] temp = head.Split(' ');
            int total = Convert.ToInt32(temp[0]);
            Global.jiange = Convert.ToInt32(temp[1]);
            Global.fps = Convert.ToInt32(temp[2]);
            Global.eventList.Clear();
            for (int i = 0; i < total; i++)
            {
                EventNode node = new EventNode();
                node.trackList = new List<Rectangle>();
                string line = fileReader.ReadLine();
                string [] data = line.Split(' ');
                node.startFrame = Convert.ToInt32(data[0]);
                node.endFrame = Convert.ToInt32(data[1]);
                int trackNum = Convert.ToInt32(data[2]);
                for (int j = 0; j < trackNum; j++)
                {
                    int x = Convert.ToInt32(data[j * 4 + N]);
                    int y = Convert.ToInt32(data[j * 4 + N + 1]);
                    int width = Convert.ToInt32(data[j * 4 + N + 2]);
                    int height = Convert.ToInt32(data[j * 4 + N + 3]);
                    Rectangle r = new Rectangle(x, y, width, height);
                    node.trackList.Add(r);
                }
                Global.eventList.Add(node);
            }
            fileReader.Close();
        }
    }
}

File: VideoAbstract_QTver/VideoAbstract_QTver/VideoAbstract_QTver/FileOperation.cpp
#include "FileOperation.h"


FileOperation::FileOperation(void)
{
}


FileOperation::~FileOperation(void)
{
}

void FileOperation::writeToFile(char* filePath, int jiange, int fps, int key_jiange, vector<EventNode> eventList)
{
	
	FILE*fs = fopen(filePath, "w+");
	fprintf(fs, "%d %d %d %d\n", eventList.size(), jiange, fps, key_jiange);
	for (int i = 0; i < eventList.size(); i++)
	{
		EventNode node = eventList[i];
		int startFrame = node.startFrame;
		int endFrame = node.endFrame;
		fprintf(fs, "%d %d %d ", startFrame, endFrame, node.trackList.size());
		for (int j = 0; j < node.trackList.size(); j++)
		{
			int x = node.trackList[j].x;
			int y = node.trackList[j].y;
			int width = node.trackList[j].width;
			int height = node.trackList[j].height;
			fprintf(fs, "%d %d %d %d ", x, y, width, height);
		}
		fprintf(fs, "\n");
	}
	//fflush(fs);
	fclose(fs);
}

void FileOperation::readFromFile(QString filePath, VideoAnalyze*&thread)
{
	int N = 3;
	//QTextCodec *code = QTextCodec::codecForName("GBK");
	QFile file(filePath);
	file.open(QFile::ReadOnly);
	QTextStream stream(&file);

	QString lineString;
	QStringList data;

	lineString = stream.readLine();
	data = lineString.split(" ");
	int total = data[0].toInt();
	thread->jiange = data[1].toInt();
	thread->fps = data[2].toInt();
	thread->key_jiange = data[3].toInt();
	thread->eventList.clear();

	for (int i = 0; i < total; i++)
	{
		EventNode node;
		QString line = stream.readLine();
		QStringList data = line.split(' ');
		node.startFrame = data[0].toInt();
		node.endFrame = data[1].toInt();
		int trackNum = data[2].toInt();
		for (int j = 0; j < trackNum; j++)
		{
			int x = data[j * 4 + N].toInt();
			int y = data[j * 4 + N + 1].toInt();
			int width = data[j * 4 + N + 2].toInt();
			int height = data[j * 4 + N + 3].toInt();
			Rect r(x, y, width, height);
			node.trackList.push_back(r);
		}
		thread->eventList.push_back(node);
	}
}